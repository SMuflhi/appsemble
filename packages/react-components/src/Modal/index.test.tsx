import { shallow } from 'enzyme';
import React from 'react';

import { Modal } from '.';

jest.mock('react-intl', () => {
  const reactIntl = jest.requireActual('react-intl');
  const intl = reactIntl.createIntl({
    locale: 'en',
  });

  return {
    ...reactIntl,
    useIntl: () => intl,
  };
});

it('should not render a bulma modal when it is inactive', () => {
  const wrapper = shallow(<Modal isActive={false}>test</Modal>);
  expect(wrapper).toMatchSnapshot();
});

it('should render a bulma modal when it is active', () => {
  const wrapper = shallow(<Modal isActive>test</Modal>);
  expect(wrapper).toMatchSnapshot();
});

it('should close the modal when the close button is clicked', () => {
  const onClose = jest.fn();
  const wrapper = shallow(
    <Modal isActive onClose={onClose}>
      test
    </Modal>,
  );
  wrapper.find('.delete').simulate('click');
  expect(onClose).toHaveBeenCalledWith();
});

it('should close the modal when the background is clicked', () => {
  const onClose = jest.fn();
  const wrapper = shallow(
    <Modal isActive onClose={onClose}>
      test
    </Modal>,
  );
  wrapper.find('.modal-background').simulate('click');
  expect(onClose).toHaveBeenCalledWith();
});

it('should close the modal escape is pressed on the background', () => {
  const onClose = jest.fn();
  const wrapper = shallow(
    <Modal isActive onClose={onClose}>
      test
    </Modal>,
  );
  wrapper.find('.modal-background').simulate('keydown', { key: 'Escape' });
  expect(onClose).toHaveBeenCalledWith(expect.any(Object));
});

it('should not close the modal another key is pressed on the background', () => {
  const onClose = jest.fn();
  const wrapper = shallow(
    <Modal isActive onClose={onClose}>
      test
    </Modal>,
  );
  wrapper.find('.modal-background').simulate('keydown', { key: 'Space' });
  expect(onClose).not.toHaveBeenCalled();
});
