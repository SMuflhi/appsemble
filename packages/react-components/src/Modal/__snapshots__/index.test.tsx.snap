// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render a bulma modal when it is inactive 1`] = `""`;

exports[`should render a bulma modal when it is active 1`] = `
<div
  className="is-active modal root open"
>
  <div
    className="modal-background"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
  />
  <div
    className="modal-card"
  >
    <div
      className="modal-card-head"
    >
      <p
        className="modal-card-title"
      />
      <button
        className="delete is-large"
        onClick={[Function]}
        type="button"
      />
    </div>
    <div
      className="modal-card-body"
    >
      test
    </div>
  </div>
</div>
`;
